EmployeeTaskSolicy/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,120 @@
+            }
+        }
+
+        [HttpDelete]
+        [Route("DeleteEmployee")]
+        public async Task<IActionResult> DeleteEmployee([FromBody] EmployeeDto employee)
+        {
+            try
+            {
+                int id;
+                if(employee.Id == null)
+                    return BadRequest("Input Employee's Id,which you want to delete from DB");
+                if (employee.Age != null || employee.FirstName != null || employee.Country != null || employee.LastName != null)
+                    return BadRequest("Please input only Employee's Id,which you want to delete from DB                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ");
+                id = (int)employee.Id;
+                var result = await _employeeRepo.DeleteEmployee(id);
+                return Ok(result);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(500, ex.Message);
+            }
+        }
+
+        [HttpPut]
+        [Route("UpdateEmployee")]
+        public async Task<IActionResult> UpdateEmployee([FromBody] EmployeeDto employee)
+        {
+            try
+            {
+                int id, age;
+                if (employee.Id == null || employee.Age == null || employee.FirstName == null || employee.Country == null || employee.LastName == null)
```

Comment:
This logic is what Validator means. You need to Write another class or method and use `bool ValidateEmployee(Employee e)`  function to make sure the data is valid. So you can avoid code duplication in different functions and will call one function in all places. 

There are frameworks that do this instead of you. For example [FluentValidator](https://docs.fluentvalidation.net/en/latest/start.html)